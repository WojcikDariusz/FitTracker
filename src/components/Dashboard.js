import React, { Component } from 'react';

class Dashboard extends Component {

    render() {
        return (
        <div className='dashboard col-xs-8'>
            <div id="myDashHeader">
                <center>
                    <h2>
                        My dashboard
                    </h2>
                </center>
            </div>
        </div>
        )
    };
};
export default Dashboard;