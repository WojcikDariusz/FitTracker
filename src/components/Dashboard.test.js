import React from 'react';
import ReactDOM from 'react-dom';
import Dashboard from './Dashboard';
import {configure,shallow} from 'enzyme';
import adapter from 'enzyme-adapter-react-16';

configure({adapter: new adapter()});

describe('Dashboard tests', () => {

    it('Correctly renders the dashboard html', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Dashboard />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('Snapshot matches', () => {
        const wrapper = shallow(<Dashboard />);
        expect(wrapper).toMatchSnapshot();
    });
});

