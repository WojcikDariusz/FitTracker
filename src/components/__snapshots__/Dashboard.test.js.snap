// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard tests Snapshot matches 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dashboard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        id="myDashHeader"
      >
        <center>
          <h2>
            My dashboard
          </h2>
        </center>
      </div>,
      "className": "dashboard col-xs-8",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <center>
          <h2>
            My dashboard
          </h2>
        </center>,
        "id": "myDashHeader",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h2>
            My dashboard
          </h2>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "My dashboard",
          },
          "ref": null,
          "rendered": "My dashboard",
          "type": "h2",
        },
        "type": "center",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          id="myDashHeader"
        >
          <center>
            <h2>
              My dashboard
            </h2>
          </center>
        </div>,
        "className": "dashboard col-xs-8",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <center>
            <h2>
              My dashboard
            </h2>
          </center>,
          "id": "myDashHeader",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h2>
              My dashboard
            </h2>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "My dashboard",
            },
            "ref": null,
            "rendered": "My dashboard",
            "type": "h2",
          },
          "type": "center",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
